src/Chunk.ts

```diff
@@ -405,19 +355,46 @@ export default class Chunk {
 
 		// external star exports
 		if (name[0] === '*') {
-			return { name: '*', module: this.graph.moduleById.get(name.substr(1)) };
+			return { variable: undefined, module: this.graph.moduleById.get(name.substr(1)) };
 		}
 
 		// resolve known star exports
 		for (let i = 0; i < module.exportAllModules.length; i++) {
 			const exportAllModule = module.exportAllModules[i];
 			// we have to ensure the right export all module
 			if (exportAllModule.traceExport(name)) {
-				return this.traceExport(exportAllModule, name);
+				return this.traceExport(name, exportAllModule);
 			}
 		}
 	}
 
+	getVariableExportName(variable: Variable) {
+		for (let exportName of Object.keys(this.exportNames)) {
+			if (this.exportNames[exportName] === variable) return exportName;
+		}
+	}
+
+	generateInternalExports(mangle: boolean = false) {
+		if (this.isEntryModuleFacade) return;
+		let i = 0,
+			safeExportName: string;
+		this.exportNames = {};
+		for (let variable of Array.from(this.exports.keys())) {
+			if (mangle) {
+				do {
+					safeExportName = (i++).toString(36);
+				} while (safeExportName.charCodeAt(0) > 47 && safeExportName.charCodeAt(0) < 58); // 0-9
+			} else {
+				i = 0;
+				safeExportName = variable.name;
+				while (this.exportNames[safeExportName]) {
+					safeExportName = variable.name + '$' + ++i;
+				}
+			}
+			this.exportNames[safeExportName] = variable;
```

Comment:
Beautiful! I hoped there was an easy solution like this!

Reply:
Yup, this was a worthwhile refactoring to wait on! Good to follow the gut on these things and not always merge the naive approach :)